dhis-2/dhis-services/dhis-service-schema/src/main/java/org/hisp/dhis/schema/introspection/JacksonPropertyIntrospector.java

```diff
@@ -357,20 +353,14 @@ private static List<Property> collectProperties(Class<?> klass) {
         continue;
       }
 
-      Property property = new Property(klass, method, null);
+      Class<?> type = method.getReturnType();
+      Property property = new Property(type, method, null);
```

Comment:
FYI: Same here, `klass` was the wrong type but that was overridden later with correct type.

Reply:
Would be to replace the "klass" naming with "type" consistently. Spring uses "type" which I guess is more common and descriptive.

Reply:
Renamed now